src/FormField/DatePickerField.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import { props, t, ReactChild } from '../utils';
+import * as cx from 'classnames';
+import View from 'react-flexview';
+import DatePicker from '../DatePicker';
+import { FormField } from './FormField';
+import { ObjectOmit } from 'typelevel-ts';
+
+export namespace DatePickerField {
+  export type Props<T extends DatePicker.Value> = {
+    /** the label for the field */
+    label: JSX.Element | string,
+    /** whether the field is required */
+    required?: boolean,
+    /** optional props to pass to the wrapping View */
+    viewProps?: View.Props,
+    /** an optional class name to pass to top level element of the component */
+    className?: string,
+    /** an optional style object to pass to top level element of the component */
+    style?: React.CSSProperties,
+    /** an optional id passed to the input component */
+    id?: string
+  } & ObjectOmit<DatePicker.Props<T>, 'returnFormat'>;  /* returnFormat is causing issues with the inference of OnChangeProps */
```

Comment:
Since TS 2.6.2 is not able to correctly infer the type for `onChangeProps` (union type for `returnFormat` and `onChange`), I dropped the support for one of the two APIs (it won't be possible to specify a `returnFormat` and receive the new value as string in `onChange`).
Newer versions of TS should be able to infer this correctly, but they break other things.